fem/libceed/ceed-assemble.cpp

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "ceed-assemble.hpp"
+
+#ifdef MFEM_USE_CEED
+#include "ceedsolvers-utility.h"
+
+int CeedHackReallocArray(size_t n, size_t unit, void *p)
+{
+   *(void **)p = realloc(*(void **)p, n*unit);
+   if (n && unit && !*(void **)p)
+      // LCOV_EXCL_START
+      return CeedError(NULL, 1, "realloc failed to allocate %zd members of size "
+                       "%zd\n", n, unit);
+   // LCOV_EXCL_STOP
+
+   return 0;
+}
+
+#define CeedHackRealloc(n, p) CeedHackReallocArray((n), sizeof(**(p)), p)
+
+int CeedHackFree(void *p)
+{
+   free(*(void **)p);
+   *(void **)p = NULL;
+   return 0;
+}
+
+namespace mfem
+{
+
+int CeedSingleOperatorFullAssemble(CeedOperator op, SparseMatrix *out)
```

Comment:
@jedbrown @jeremylt Don't you think it would be good to have this directly in libCEED?